go.mod

```diff
@@ -1,6 +1,6 @@
 module github.com/prometheus/prometheus
 
-go 1.20
+go 1.21
```

Comment:
That's not valid anymore with 1.21.x, see https://github.com/golang/go/issues/62278#issuecomment-1693538776

Reply:
Specified version to 1.21.2

Reply:
This fails the some tests since. 1.21.x format is not supported.
`/__w/prometheus/prometheus/go.mod:3: invalid go version '1.21.2': must match format 1.23`